components/test-suite/src/loci/tests/testng/FormatReaderTestFactory.java

```diff
@@ -223,6 +223,11 @@ public String getProperty(String key) {
       String file = fileSet.iterator().next();
       try {
         reader.setId(file);
+      } catch (Exception e) {
+        LOGGER.error("setId({}) failed", file, e);
```

Comment:
Tested in https://ci.openmicroscopy.org/job/BIOFORMATS-DEV-merge-repository-subset/354/console /cc @melissalinkert 

As discussed, assuming this does not introduce more instability, introducing a fail-fast mechanism for files failing `setId()` might make sense and save unnecessary CPU.
One RFE to would be to catch all files failing `setId`, captures them in a list and then throw a `RuntimeException`.